### PR TITLE
Prevents sending empty tokens to intercom

### PR DESCRIPTION
**Pull request by obrunsmann**
Referring to issue #102 it's very dangerous that the app fully crashes if the token is nil. I was not able to reproduce, why `getToken()` of Firebase Messaging plugin returns null but when callng `Intercom.sendTokenToIntercom(null)` I was able to reproduce the crash. 

So I suggest adding any check which prevents sending non strings to the native handler. While we are working with dynamic return type, I added an exception. I could also imagine to work with boolean return type while I now preferred not changing the function's signature.

### CODE REVIEW

File: lib/intercom_flutter.dart [hunk 1]
```diff
@@ -119,6 +119,13 @@ class Intercom {
   }
 
   static Future<dynamic> sendTokenToIntercom(String token) {
+    assert(token?.isNotEmpty == true);
```
Comment (Zazo032) [inline lib/intercom_flutter.dart:122]:
```suggestion
    assert(token != null && token.isNotEmpty);
```

File: lib/intercom_flutter.dart [hunk 2]
```diff
@@ -119,6 +119,13 @@ class Intercom {
...
+
+    // empty token let the app crash, so prevent before invoking method
+    if (token?.isNotEmpty == false) {
+      throw 'Push token must not be empty.';
+    }
```
Comment (Zazo032) [inline lib/intercom_flutter.dart:127]:
```suggestion
    // A null or empty token may crash the app
    if (token == null || token?.isEmpty) {
      throw '[intercom_flutter] Push token must not be null or empty.';
    }
```

**Review by Zazo032 — CHANGES_REQUESTED** [review]
Hi @obrunsmann! Overall changes LGTM. Only a few small nits about conditions.

Also, do you think you could add a test that verifies that `sendTokenToIntercom` asserts or throws if calling with a `null` or empty token? Thanks!

**Comment by deepak786** [comment]
As the plugin is now null safe. So the null token cannot be passed. 
For the empty token, there is already an assertion.
https://github.com/v3rm0n/intercom_flutter/blob/359663fb2e49103b501328a3a8c1bcd6a1adca02/intercom_flutter_platform_interface/lib/method_channel_intercom_flutter.dart#L134

So closing the pull request. 
If you still have an issue, feel free to open a new issue or submit the pull request.
Thanks.
